internal/tempfile/tempfile_linux.go

```diff
@@ -8,9 +8,11 @@ import (
 	"golang.org/x/sys/unix"
 )
 
+const permissions = 0o600
+
 // Create creates a temporary file that will be automatically deleted on close.
-func Create(dir string) (*os.File, error) {
-	dir = tempDirOr(dir)
+func CreateAutoDelete() (*os.File, error) {
```

Comment:
Removed the `dir` parameter based on @jkowalski suggestion

Reply:
And renamed the function to `CreateAutoDelete` to make it more explicit.
@jkowalski @Rohit-BM18 @redgoat650 if you have a suggestion for a better name, we can change it in a separate PR.